src/lib/block_producer/block_producer.ml

```diff
@@ -541,15 +551,13 @@ module Vrf_evaluation_state = struct
     poll ~logger ~vrf_evaluator t
 end
 
-let run ~logger ~vrf_evaluator ~prover ~verifier ~trust_system
-    ~get_completed_work ~transaction_resource_pool ~time_controller
-    ~consensus_local_state ~coinbase_receiver ~frontier_reader
+let run ~context:(module Context : CONTEXT) ~vrf_evaluator ~prover ~verifier
```

Comment:
Small NIT: would the following pattern be preferable instead?

```ocaml
let f ~context ... =
  let open (val context : CONTEXT) in
  ...
```

The advantage of this is that we can just call `~context` whenever we pass the context to a child, instead of having to write `~context:(module Context)` each time. I haven't tested this, but I think OCaml should still be able to reason about the module signature type conversion correctly with this pattern.

Reply:
I chose the current pattern because OCaml will attempt to unify the two context module types, preventing us from contacting or expanding the context.